src/annotator/pdf-sidebar.js

```diff
@@ -12,8 +16,101 @@ const defaultConfig = {
   },
 };
 
+// In accordance with the minimum width PDFJS asserts for its #mainContainer
+// element, we shouldn't try to shrink down the PDF container any narrower
+// than this value when resizing it in side-by-side mode. If the available space
+// left in the viewport after alloting space for the sidebar is less than this
+// value, don't apply side-by-side mode, but instead allow the sidebar to
+// overlap the PDF
+const MIN_PDF_WIDTH = 320;
+
 export default class PdfSidebar extends Sidebar {
   constructor(element, config) {
     super(element, { ...defaultConfig, ...config });
+
+    this.lastSidebarLayoutState = {
+      expanded: false,
+      width: 0,
+      height: 0,
+    };
+
+    this.window = /** @type {HypothesisWindow} */ (window);
+    this.pdfViewer = this.window.PDFViewerApplication?.pdfViewer;
+    this.pdfContainer = this.window.PDFViewerApplication?.appConfig?.appContainer;
+
+    // Prefer to lay out the sidebar and the PDF side-by-side (with the sidebar
+    // not overlapping the PDF) when space allows
+    this.sideBySide =
+      config.enableExperimentalPDFSideBySide &&
+      this.pdfViewer &&
+      this.pdfContainer;
+
+    // Is the current state of the layout side-by-side?
+    this.sideBySideActive = false;
+
+    if (this.sideBySide) {
+      this.subscribe('sidebarLayoutChanged', state =>
+        this.fitSideBySide(state)
+      );
+      window.addEventListener('resize', () => this.fitSideBySide());
+    }
+  }
+
+  /**
+   * Override `Guest.onElementClick` to prevent the sidebar from being
+   * hidden on document clicks if in side-by-side layout
+   */
+  onElementClick(event) {
```

Comment:
Rather than overriding the `onElementClick` method and then duplicating some of the logic from `Guest` to send the `hideSidebar` message, I think it would be preferable to have either a property or overridable method on the `Guest` class which determines whether auto-collapsing of the sidebar is enabled or not. If using a property, this class would update it when `this.sideBySideActive` changes. Iif using an overridable method, the `Guest` class would call it and it would return the current value of `this.sideBySideActive`.

The idea is to keep the responsibility of actually hiding the sidebar in the `Guest` class and just delegate the decision of whether or not to apply it to the subclass.

Reply:
There were some comments in `Guest` saying that the code should really live in `Sidebar`, so I was trying to follow that design direction. I can indeed flip it around!

Reply:
Also, just to save me a tiny bit of pain: you're not currently working on converting `Guest` to JS from coffee right now, are you?

Reply:
No, I'm not working on that.

> There were some comments in Guest saying that the code should really live in Sidebar, so I was trying to follow that design direction. I can indeed flip it around!

Hmm, well perhaps it _would_ make sense for the logic to live in `Sidebar` rather than `Guest`, but more importantly it should be in just one place rather than two.

Reply:
I agree it should be one place... I wasn't thrilled with it, as I mentioned in https://github.com/hypothesis/client/pull/2484#pullrequestreview-477000255

For now, I can put it in `Guest`. Once we've converted the full tree to JS, we should reassess, perhaps?

Reply:
Yep, that sounds reasonable.